src/plugin.rs

```diff
@@ -83,9 +85,12 @@ impl AnalyzerPlugin for CairoLint {
             .filter(|diag| {
                 let node = diag.stable_ptr.lookup(db.upcast());
                 let allowed_name = get_name_for_diagnostic_message(&diag.message).unwrap();
-                let lint_type = get_lint_type_from_diagnostic_message(&diag.message);
+                let default_allowed = is_lint_enabled_by_default(&diag.message).unwrap();
```

Comment:
I don't understand why the diagnostics are filtered here in such way - is it not possible to filter those earlier?

Reply:
We could filter out some checking functions, but it won't guarantee that this config will be honored. Single checking function can be used to check multiple lints. So there is a case where:

We got lints A and B, both checked by checker C. 
Case 1:
- Config says we want to ignore lint A and B. We just skip checker C, and skipping is guaranteed.
Case 2:
- Config says we want to ignore lint A only. We cannot skip checker C as we still want to include lint B. So we cannot  guarantee that the lint A will be skipped if we operate earlier on the checkers.

At most what we can do is filter out the first case, but i don't think it's that necessary, as it won't boost the performence that much, and will make the plugin code more confusing.

WDYT?

Reply:
discussed offline